acnportal/acnsim/events/event.py

```diff
@@ -120,48 +118,33 @@ def _from_dict(
         return out_obj, loaded_dict
 
 
-class UnplugEvent(Event):
+class PluginEvent(EVEvent):
+    """ Subclass of Event for EV plugins.
+
+    Args:
+        timestamp (int): See Event.
+        ev (EV): The EV which will be plugged in.
+    """
+
+    def __init__(self, timestamp, ev):
+        super().__init__(timestamp, ev)
+        self.event_type = "Plugin"
+        self.precedence = 10
+
+
+class UnplugEvent(EVEvent):
     """ Subclass of Event for EV unplugs.
 
     Args:
         timestamp (int): See Event.
-        station_id (str): ID of the EVSE where the EV is to be unplugged.
-        session_id (str): ID of the session which should be ended.
+        ev (EV): The EV which will be unplugged.
     """
 
-    def __init__(self, timestamp, station_id, session_id):
-        super().__init__(timestamp)
+    def __init__(self, timestamp, ev):
+        super().__init__(timestamp, ev)
         self.event_type = "Unplug"
-        self.station_id = station_id
-        self.session_id = session_id
         self.precedence = 0
 
-    def _to_dict(
-        self, context_dict: Optional[Dict[str, Any]] = None
-    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
-        """ Implements BaseSimObj._to_dict. """
-        attribute_dict, context_dict = super()._to_dict(context_dict)
-        # Unplug-specific attributes
-        attribute_dict["station_id"] = self.station_id
-        attribute_dict["session_id"] = self.session_id
-        return attribute_dict, context_dict
-
-    @classmethod
-    def _from_dict(
```

Comment:
Hmm. That is annoying. The only time Unplug events are serialized and deserialized is in the event history... I wonder if we can make an UnplugEvent_v1 or something like that solely for that purpose. We don't have to support processing it since all unplug events (at least in our simulations) are generated dynamically from PlugInEvents.

Reply:
One possibility is writing logic to convert old-version UnplugEvents into new ones within the deserializer logic itself. The result will be a new-version UnplugEvent with an EV with an incomplete definition (no arrival, departure, energy req), but we can put out a warning about this during deserialization. Since in theory breaking changes like this should be rare, I'm okay with making a one-off change to the serializer for this case. @zach401 how does that sound to you?

Reply:
If it's not too hard of a change it might be worth it. Do you want to make the change or should I?

Reply:
I will